lib/src/list_extensions.dart

```diff
@@ -241,7 +241,7 @@ extension ListExtensions<E> on List<E> {
   ListSlice<E> slice(int start, [int? end]) {
     end = RangeError.checkValidRange(start, end, length);
     var self = this;
-    if (self is ListSlice) return self.slice(start, end);
+    if (self is ListSlice<E>) return self.slice(start, end);
```

Comment:
(This is a constant foot-gun. I'll add another notch to the issue suggesting we promote to `ListSlice<E>` here.)